screens/GameDetails.js

```diff
@@ -0,0 +1,301 @@
+    if (player.id == details.organizer) {
+      return player.name;
+    } else {
+      const organizer = await DataStore.query(Player, details.organizer);
+      console.log("organizer returned: ", organizer);
+      return organizer.name;
+    }
+  }
+
+
+  useEffect(() => {
+    async function fetchData() {
+      await getPlayer();
+      await getGameOrganizer();
+      await getInvitedPlayers();
+    }
+  
+    fetchData();
+  }, []);
+  
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        setLoading(true);
+  
+        const playerRes = await getPlayer();
+        setThisPlayer(playerRes);
+  
+        const organizerRes = await getGameOrganizer(playerRes);
+        setGameOrganizer(organizerRes);
+  
+        const invitedPlayersRes = await getInvitedPlayers();
+        setAccepted(invitedPlayersRes.accepted);
+        setDeclined(invitedPlayersRes.declined);
+  
+        setLoading(false);
+      } catch (error) {
+        console.log("Error fetching data: ", error);
+        setLoading(false);
+      }
+    }
+  
+    fetchData();
+  }, []);
+
+
+  function isGameOwner() {
+    // Use a dummy username until actual organizer names are used
+    // We would need DUMMY_USERNAME to be the name of the current user
+    return thisPlayer.id == details.organizer;
+  }
+
+  function showDescription() {
+    return details.description != "" && details.description != undefined;
+  }
+
+  const renderItem = ({ item, index }) => {
+    return (
+      <View key={index} style={styles.nameContainer}>
+        <Text style={styles.text}>{item.name}</Text>
+        <Text
+          style={[
+            styles.text,
+            item.status == "In" ? { color: "green" } : { color: "red" },
+          ]}
+        >
+          {item.status}
+        </Text>
+      </View>
+    );
+  };
+
+  function handleEdit() {
+    navigation.navigate(("UpdateGame"), {
+        game: route.params.item,
+      });
+  }
+ 
+  async function handleDelete() {
+    setLoading(true);
+    const toDelete = await DataStore.query(Game, details.id);
+    DataStore.delete(toDelete);
+    setLoading(false);
+    navigation.navigate("HomeScreen");
+  }
+
+  return (
+    <View style={styles.container}>
+      <BackArrow location="HomeScreen" />
+      <View style={styles.infoContainer}>
+        <Text style={styles.topText}>{details.name}</Text>
+        <Text style={styles.text}>
+          <Text style={styles.bold}>Date: </Text>
+          {epochToLocalDate(details.datetime)}
+        </Text>
+        <Text style={styles.text}>
+          <Text style={styles.bold}>Location: </Text>
+          {details.location}
+        </Text>
+        <Text style={styles.text}>
+          <Text style={styles.bold}>Time: </Text>
+          {epochToLocalTime(details.datetime)}
+        </Text>
+        <Text style={styles.text}>
+          <Text style={styles.bold}>Organizer: </Text>
+          {loading ? "Loading..." : gameOrganizer}
```

Comment:
Do we not get the organizer from the original fetch for data on the HomeScreen? I thought it would already be in the game details. Also, use a LoadingScreen instead of this.

Reply:
No, we do not already get the organizer - the id only is part of the game object, we have to retrieve it once inside game details.